src/include/sof/audio/volume_uuid.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Baofeng Tian <baofeng.tian@intel.com>
+ */
+
+#ifndef __SOF_AUDIO_VOLUME_UUID_H__
+#define __SOF_AUDIO_VOLUME_UUID_H__
+
+#include <sof/lib/uuid.h>
+#include <sof/trace/trace.h>
+
+#if CONFIG_IPC_MAJOR_3
+/* b77e677e-5ff4-4188-af14-fba8bdbf8682 */
+DECLARE_SOF_RT_UUID("pga", volume_uuid, 0xb77e677e, 0x5ff4, 0x4188,
+		    0xaf, 0x14, 0xfb, 0xa8, 0xbd, 0xbf, 0x86, 0x82);
+#else
+/* these ids aligns windows driver requirement to support windows driver */
+/* 8a171323-94a3-4e1d-afe9-fe5dbaa4c393 */
+DECLARE_SOF_RT_UUID("pga", volume_uuid, 0x8a171323, 0x94a3, 0x4e1d,
+		    0xaf, 0xe9, 0xfe, 0x5d, 0xba, 0xa4, 0xc3, 0x93);
+
+/* 61bca9a8-18d0-4a18-8e7b-2639219804b7 */
+DECLARE_SOF_RT_UUID("gain", gain_uuid, 0x61bca9a8, 0x18d0, 0x4a18,
+		    0x8e, 0x7b, 0x26, 0x39, 0x21, 0x98, 0x04, 0xb7);
+
+DECLARE_TR_CTX(gain_tr, SOF_UUID(gain_uuid), LOG_LEVEL_INFO);
+#endif
+
+DECLARE_TR_CTX(volume_tr, SOF_UUID(volume_uuid), LOG_LEVEL_INFO);
```

Comment:
This is a bit iffy to use use these DECLARE_ macros in a header file. If this is ever included twice, you'll get errors.

Did you consider having these in volume_ipc3.c and volume_ipc4.c ? You'd have to duplicate the "volume_tr" duplication, but otherwise this should work. Any other downsides?

Reply:
Ack - @btian1 can we look at moving the volume headers to the volume src directory (in a new PR) as AFAICT they all should be private for volume only (except for any IPC structures).

Reply:
@kv2019i , move it to ipc3.c and ipc4.c will be like my previous implementation, volume.c will refer these declaration, it is hard to refer.
 I have to create new header file to  accommodate this and make sure it only be included once.

@lgirdwood , I also want to move each header file to each folder, however:

1. all the component directory does not include headers.
2. how to resolve cmocka include? like in: 
    sof\test\cmocka\src\audio\volume\volume_process.c
    #include <sof/audio/volume.h>

if we want to move, we need take care each component, not only volume and resolve issue 2.